openpype/hooks/pre_ocio_hook.py

```diff
@@ -0,0 +1,60 @@
+from openpype.lib import PreLaunchHook
+
+from openpype.pipeline.colorspace import (
+    get_imageio_config,
+    is_host_use_ocio_config_activated
+)
+from openpype.pipeline.template_data import get_template_data_with_names
+
+
+class OCIOEnvHook(PreLaunchHook):
+    """Set OCIO environment variable for hosts that use OpenColorIO."""
+
+    order = 0
+    app_groups = [
+        "fusion",
+        "blender",
+        "aftereffects",
+        "3dsmax",
+        "houdini",
+        "maya",
+        "nuke",
+        "nukex",
+        "nukeassist",
+        "nukestudio",
+        "hiero"
+    ]
+
+    def execute(self):
+        """Hook entry method."""
+
+        template_data = get_template_data_with_names(
+            project_name=self.data["project_name"],
+            asset_name=self.data["asset_name"],
+            task_name=self.data["task_name"],
+            host_name=self.host_name,
+            system_settings=self.data["system_settings"]
+        )
+
+        config_data = get_imageio_config(
+            project_name=self.data["project_name"],
+            host_name=self.host_name,
+            project_settings=self.data["project_settings"],
+            anatomy_data=template_data,
+            anatomy=self.data["anatomy"]
+        )
+
+        if config_data:
+            use_config_path = is_host_use_ocio_config_activated(
+                project_name=self.data["project_name"],
+                host_name=self.host_name,
+                project_settings=self.data["project_settings"]
+            )
```

Comment:
Isn't it weird that if the host is somehow explicitly disabling the use OCIO config setting that `get_imageio_config` still returns a data for that host? Not sure what's going on here.

Reply:
This should not happen that is why there is `if config_data:` on the scene. If the host is disabling Colorspace management then `get_imageio_config` should return None. Or do you see anywhere in the code it is different?

Reply:
Again, same question. If the host color management is enabled I am still not sure why we need to check 'use ocio config'. If it is color managed then it is using OCIO. If it is disabled it's not using OCIO (or at least not in a way that OpenPype should be handling it?)

Reply:
Again, same answer. Colormanagement activated is not the same as using OCIO config. You don't need to use our OCIO config to be able use our color management. Someone might want to use DCC's internal color management. We need to be explicit abou this one since those DCC's are also explicit. 

Without `use_ocio_config` someone might wonder - in DCC like Flame which is not supported custom OCIO config path,  where is the color config used from.

Reply:
Should we maybe name it `set_ocio_config` instead of `use_ocio_config`, because if color management is enabled in e.g. maya then it would still have a an OCIO config. We would just not be setting it?

Reply:
+1 on `set_ocio_config`

Reply:
Done ;)